### PR TITLE
fix: cleanup and delete ea crs if kubeconfigs are not found

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- with this PR we would remove EA-CRs if:
  EA-CR exists
  no kubeconfig is found for the cluster
  no kyma-CR is found for the cluster

**Related issue(s)**

### CODE REVIEW

File: controllers/kyma_controller_test.go [hunk 1]
```diff
@@ -3,59 +3,59 @@ package controllers_test
...
+// 	Context("Creating Kyma CR", func() {
+// 		It("should create Kyma CR", func() {
+// 			kyma = createKymaResource(crName)
+//
+// 			verifyEventingAuth(kyma.Namespace, kyma.Name)
+//
+// 			deleteKymaResource(kyma)
+// 		})
+// 	})
+// })
```
Comment (marcobebway) [inline controllers/kyma_controller_test.go:58]: Can we delete this?

File: controllers/eventingauth_controller_test.go [hunk 2]
```diff
@@ -305,8 +324,18 @@ func deleteKubeconfigSecret(crName string) {
...
+	}, defaultTimeout).Should(Succeed())
+}
+
+func reconcileEventingAuth(cr *eamapiv1alpha1.EventingAuth) {
+	By("Labelling EA to force reconcilation")
+	Eventually(func(g Gomega) {
+		var ea eamapiv1alpha1.EventingAuth
+		err := k8sClient.Get(context.TODO(), kpkgclient.ObjectKeyFromObject(cr), &ea)
+		g.Expect(err).To(BeNil())
+		ea.ObjectMeta.Labels = map[string]string{"reconcile": "now"}
```
Comment (marcobebway) [inline controllers/eventingauth_controller_test.go:337]: With the same label key/value the reconciliation will be triggered once, is this the intended behavior?

File: controllers/eventingauth_controller.go [hunk 3]
```diff
@@ -98,6 +101,13 @@ func (r *eventingAuthReconciler) Reconcile(ctx context.Context, req kcontrollerr
 func (r *eventingAuthReconciler) handleApplicationSecret(ctx context.Context, logger logr.Logger, cr eamapiv1alpha1.EventingAuth) (kcontrollerruntime.Result, error) {
 	skrClient, err := skr.NewClient(r.Client, cr.Name)
 	if err != nil {
+		if errors.Is(err, skr.ErrSecretNotFound) {
+			logger.Error(err, "Failed to load kubeconfig for cluster", "clusterID", cr.Name)
+			if !r.kymaCRExists(ctx, cr.Name) {
```
Comment (marcobebway) [inline controllers/eventingauth_controller.go:106]: 3 nested ifs, can you please simplify?

File: controllers/eventingauth_controller.go [hunk 4]
```diff
@@ -161,6 +171,15 @@ func (r *eventingAuthReconciler) handleApplicationSecret(ctx context.Context, lo
 	return kcontrollerruntime.Result{}, nil
 }
 
+func (r *eventingAuthReconciler) kymaCRExists(ctx context.Context, clusterID string) bool {
+	var kyma klmapiv1beta2.Kyma
+	err := r.Get(ctx, types.NamespacedName{Namespace: skr.KcpNamespace, Name: clusterID}, &kyma)
+	if kapierrors.IsNotFound(err) {
+		return false
+	}
+	return true
```
Comment (marcobebway) [inline controllers/eventingauth_controller.go:180]:
```suggestion
	return !kapierrors.IsNotFound(err)
```

File: controllers/eventingauth_controller_test.go [hunk 5]
```diff
@@ -134,6 +135,15 @@ var _ = Describe("EventingAuth Controller unhappy tests", Serial, Ordered, func(
 		stubSuccessfulSkrSecretCreation()
 		verifyEventingAuthStatusReady(eventingAuth)
 	})
+
+	It("should cleanup if no kyma cr exists", func() {
+		stubSuccessfulIasAppCreation()
+		stubSuccessfulSkrSecretCreation()
+		eventingAuth = createEventingAuth(crName)
```
Comment (marcobebway) [inline controllers/eventingauth_controller_test.go:142]: Can we add one more step after `eventingAuth = createEventingAuth(crName)` to verify that it exists?

File: controllers/kyma_controller_test.go (hunk 1 above)
Comment (k15r) [reply controllers/kyma_controller_test.go:58]: oh no, that must be reactivated

File: controllers/eventingauth_controller_test.go (hunk 5 above)
Comment (k15r) [reply controllers/eventingauth_controller_test.go:142]: is implicitly covered by the next line:
`	reconcileEventingAuth(eventingAuth)` will only succeed if the EA-CR exists

**Pull request closed without merging**